packages/block-library/src/buttons/editor.scss

```diff
@@ -4,19 +4,19 @@
 }
 
 .wp-block-buttons {
-	&[data-align="center"] .block-editor-block-list__layout {
+	&[data-align="center"] > .is-block-content {
```

Comment:
Blocks starting to use this class is what scares me most, because we're unlikely to keep this helper class.